src/Umbraco.Web/Security/MembershipHelper.cs

```diff
@@ -289,11 +289,46 @@ public virtual IPublishedContent GetByProviderKey(object key)
             return MemberCache.GetByProviderKey(key);
         }
 
+        public virtual IEnumerable<IPublishedContent> GetByProviderKey(IEnumerable<object> keys)
+        {
+            return keys?.Select(GetByProviderKey).WhereNotNull() ?? new IPublishedContent[0];
```

Comment:
Instead of `?? new IPublishedContent[0];` it should be `?? Enumerable.Empty<IPublishedContent>()` to avoid allocating an object unnecessarily, same goes for the other code in this PR

Reply:
Fixed in 64bae43